refactor: preset slugs

breadadams opened the pull request.

breadadams commented:
@Kikobeats this is the initial progress of the preset slugs as we discussed, I've just used **Bytes and Humans** as an example for now.

It's all working fine, however it's not as I'd like for the `generate-previews` script. Due to the setup, the filename has to match the generated slug. That's ok for `bytes-and-humans.js`, but I would love if preset names could include special characters.

To achieve this we'd need to know the preset's `name` within the script (instead of the filename), to be able convert it to the slug.

What do you think, any suggestions?

Kikobeats commented on packages/app/src/lib/preset-slug.js:
```diff
@@ -0,0 +1,3 @@
+export const getPresetSlug = presetName => encodeURIComponent(presetName.toLowerCase().replace(/ /g, '-'))
```
I think it's safe to use `.replaceAll`

Kikobeats commented:
Looks good 👍 

I understand your point related to `generate-previews` script; what we can do is to throw an error under mismatch, that should be more than enough since still we are going to review every PR that adds a new preset.

Also, I think it's a good moment to improve the seo now you have the name preset 🙂

breadadams commented:
Thanks @Kikobeats, I made some adjustments that should resolve the potential `generate-previews` issue in the case of special characters. Instead of escaping them with `encodeURIComponent` I simply strip them. For example, the preset I've adjusted - `Next.js Conf` - would be converted to `nextjs-conf`, and therefore matching the filename.

1. Want me to add a CONTRIBUTING file with some simple instructions, including the naming?
2. Is the build failing because of `replaceAll`? I was running the site with node v12 on local and it failed, switched to v16 and all ok, but if the Vercel build is taking the version from the `engines` property it could be that.
3. What do you think about splitting the SEO into a separate task? Just to keep the PRs as bite-size as possible.

Kikobeats commented on packages/app/src/constants.js:
```diff
@@ -29,3 +29,16 @@ export const OVERLAY_STATE = {
   KEYBINDINGS: 'keybindings'
 }
+
+export const FALLBACK_SLUGS = {
```
@breadadams let's drop this, I prefer to write a network rewrite rather than keep legacy code

Kikobeats reviewed (changes requested):
just requesting to remove fallbacks since we can consider this a major version
